sdk/synapse/azure-analytics-synapse-accesscontrol/src/main/java/com/azure/analytics/synapse/accesscontrol/AccessControlClient.java

```diff
@@ -0,0 +1,265 @@
+    public List<RoleAssignmentDetails> getRoleAssignments(String roleId, String principalId, String continuationToken) {
+        return this.serviceClient.getRoleAssignments(roleId, principalId, continuationToken);
+    }
+
+    /**
+     * List role assignments.
+     * 
+     * @param roleId Synapse Built-In Role Id.
+     * @param principalId Object ID of the AAD principal or security-group.
+     * @param continuationToken Continuation token.
+     * @param context The context to associate with this operation.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorContractException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return a list of role assignments.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public List<RoleAssignmentDetails> getRoleAssignments(String roleId, String principalId, String continuationToken, Context context) {
+        return this.serviceClient.getRoleAssignments(roleId, principalId, continuationToken, context);
+    }
+
+    /**
+     * List role assignments.
+     * 
+     * @throws ErrorContractException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return a list of role assignments.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public List<RoleAssignmentDetails> getRoleAssignments() {
+        return this.serviceClient.getRoleAssignments();
+    }
+
+    /**
+     * Get role assignment by role assignment Id.
+     * 
+     * @param roleAssignmentId The ID of the role assignment.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorContractException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return role assignment by role assignment Id.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public RoleAssignmentDetails getRoleAssignmentById(String roleAssignmentId) {
+        return this.serviceClient.getRoleAssignmentById(roleAssignmentId);
+    }
+
+    /**
+     * Get role assignment by role assignment Id.
+     * 
+     * @param roleAssignmentId The ID of the role assignment.
+     * @param context The context to associate with this operation.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws ErrorContractException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return role assignment by role assignment Id.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public RoleAssignmentDetails getRoleAssignmentById(String roleAssignmentId, Context context) {
```

Comment:
Methods that accept Context in the sync APIs should follow the 'withResponse' pattern and not unwrap the Response from the return type. In this case, I would expect a Response<RoleAssignmentDetails> response type.

Reply:
Removed Context